src/h265rtpdepacketizer.cpp

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) 2023-2024 Paul-Louis Ageneau
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+#if RTC_ENABLE_MEDIA
+
+#include "h265rtpdepacketizer.hpp"
+#include "h265nalunit.hpp"
+
+#include "impl/internals.hpp"
+
+namespace rtc {
+
+const binary naluStartCode = {byte{0}, byte{0}, byte{0}, byte{1}};
```

Comment:
The start code could be selectable with a `NalUnit::Separator` parameter like with the packetizer.

Reply:
Sounds good. Just to clarify:

```cpp
/// NAL unit separator
enum class Separator {
        Length = RTC_NAL_SEPARATOR_LENGTH, // first 4 bytes are NAL unit length
        LongStartSequence = RTC_NAL_SEPARATOR_LONG_START_SEQUENCE,   // 0x00, 0x00, 0x00, 0x01
        ShortStartSequence = RTC_NAL_SEPARATOR_SHORT_START_SEQUENCE, // 0x00, 0x00, 0x01
        StartSequence = RTC_NAL_SEPARATOR_START_SEQUENCE, // LongStartSequence or ShortStartSequence
};
```

`LongStartSequence` and `ShortStartSequence` are easy, just write 00 00 00 01 or 00 00 01 into the buffer.

Should `StartSequence` just mean the same thing as `LongStartSequence`? Or `ShortStartSequence`?

`Length` is a little more complicated, we'd need to insert a placeholder for where the length should go and then fill it in later once all of the reassembled fragments have been written into the buffer and the total length can be calculated.

Or could we specify that only `LongStartSequence` and `ShortStartSequence` are supported for the value of this parameter?

Reply:
> Should `StartSequence` just mean the same thing as `LongStartSequence`? Or `ShortStartSequence`?

`StartSequence` for packetizers means the parser can expect both the long and the short start sequence, and it will effectively use both as separators. Here it should mean the same as `LongStartSequence`.

> Or could we specify that only `LongStartSequence` and `ShortStartSequence` are supported for the value of this parameter?

Yes, I think you can just throw an `invalid_argument` exception if another value is specified so you don't have to implement `Length`. I'm not sure it would be useful apart maybe for some niche use cases.

Reply:
ee96c7ade195ed0909c2909f0e843d13b8500adf